src/instructlab/sdg/utils/chunking.py

```diff
@@ -36,15 +53,82 @@ def chunk_document(documents: List, server_ctx_size, chunk_word_count) -> List[s
                 )
             )
         )
+    # Placeholder for params
     content = []
-    text_splitter = RecursiveCharacterTextSplitter(
-        separators=["\n\n", "\n", " "],
+    chunk_size = _num_chars_from_tokens(no_tokens_per_doc)
+    chunk_overlap = _DEFAULT_CHUNK_OVERLAP
+    text_splitter = RecursiveCharacterTextSplitter.from_language(
+        language=Language.MARKDOWN,
         chunk_size=_num_chars_from_tokens(no_tokens_per_doc),
         chunk_overlap=_DEFAULT_CHUNK_OVERLAP,
     )
 
+    # Determine file type for heuristics, default with markdown
     for docs in documents:
-        temp = text_splitter.create_documents([docs])
-        content.extend([item.page_content for item in temp])
+        # Try Except Block if Magika fails
+        try:
+            m = Magika()
+            docs_bytes = docs.encode("utf-8")
+            res = m.identify_bytes(docs_bytes)
+            file_type = res.output.ct_label.lower()
 
+            # Full list of supported languages of Langchain
+            # supported_types = [e.value for e in Language]
+            # Checks for file types:
+            file_type_to_language = {
+                "go": Language.GO,
+                "java": Language.JAVA,
+                "javascript": Language.JS,  # Adjusted to match the previous variable name Language.js
+                "php": Language.PHP,
+                "python": Language.PYTHON,
+                "ruby": Language.RUBY,
+                "rust": Language.RUST,
+                "latex": Language.LATEX,
+                "html": Language.HTML,
+                "cs": Language.CSHARP,
+                "c": Language.C,
+                "perl": Language.PERL,
+            }
+            if file_type in file_type_to_language:
+                language = file_type_to_language[file_type]
+                text_splitter = RecursiveCharacterTextSplitter.from_language(
```

Comment:
I would rename this so that we don't override the top-level `text_splitter` object.

Reply:
This is correct! Originally my thought process was to indicate a naive text_splitter as the base case, in `text_splitter_naive`, but my CoT got lost somewhere in the PR.